include/ThreadPool.h

```diff
@@ -57,20 +57,26 @@ class ThreadPool
 	auto enqueue(Fn&& fn, Args&&... args) -> std::future<std::invoke_result_t<Fn, Args...>>
```

Comment:
Can we leave out the argument pack parameter and only pass the function object itself? Or would that introduce unwanted restrictions on callers?

Reply:
Passing in the argument pack like this allows the arguments to be passed in directly, so stuff like this is possible:

```cpp
ThreadPool::enqueue(&foo, 1, 2, 3);
```

This will submit a task that executes `foo` with the arguments `1`, `2`, and `3`. Passing in the argument pack provides such flexibility.